File: 2ºCARRERA/tprog/practica2_872815_873338/c++/dup.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : dup.cc
 */

#include "dup.h"

void Dup::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num = pila.top();
    pila.push(num);
    pc++;
}

void Dup::muestra(){
    cout << "dup";
}

File: 2ºCARRERA/tprog/practica2_872815_873338/java/Dup.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Dup.java
 */

import java.util.Stack;

class Dup implements Instruccion
{
    public int ejecuta(Stack<Integer> pila, int pc) {
        int num;
        num = pila.peek();
        pila.push(num);
        pc++;
        return pc;
    }
    public void muestra() {
        System.out.print("dup");
    }
}

